check-mysql/lib/check-mysql.go

```diff
@@ -52,12 +62,60 @@ SubCommands:`)
 	ckr.Exit()
 }
 
-func newMySQL(m mysqlSetting) mysql.Conn {
+func newDB(m mysqlSetting) (*sql.DB, error) {
 	proto := "tcp"
 	target := fmt.Sprintf("%s:%s", m.Host, m.Port)
 	if m.Socket != "" {
 		proto = "unix"
 		target = m.Socket
 	}
-	return mysql.New(proto, "", target, m.User, m.Pass, "")
+	cfg := &mysql.Config{
+		User:                 m.User,
+		Passwd:               m.Pass,
+		Net:                  proto,
+		Addr:                 target,
+		AllowNativePasswords: true,
+	}
+
+	return sql.Open("mysql", cfg.FormatDSN())
+}
+
+func getMySQLVersion(db *sql.DB) (*mysqlVersion, error) {
+	var rawVersion string
+	err := db.QueryRow("SELECT VERSION()").Scan(&rawVersion)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to query SELECT VERSION(): %s", err)
+	}
+
+	// Version example: mysql-8.0.1-dmr
```

Comment:
afaik, value from `SELECT VERSION()` (or underlying `show variables like 'version';`) is just like `8.0.23` or `5.5.44-0+deb8u1-log` (it's ancient one!), so this example does not represent those values?

Reply:
68baa67